Desktop_Interface/uartstyledecoder.cpp

```diff
@@ -146,29 +157,30 @@ void uartStyleDecoder::decodeNextUartBit(unsigned char bitValue)
 
 //This function compensates for jitter by, when the current bit is a "1", and the last bit was a zero, setting the pointer
 //to the sample at the midpoint between this bit and the last.
-bool uartStyleDecoder::jitterCompensationProcedure(double baudRate, unsigned char current_bit){
+bool uartStyleDecoder::jitterCompensationProcedure(double baudRate, unsigned char current_bit)
+{
 
     //We only run when the current bit is a "1", to prevent slowdown when there are long breaks between transmissions.
```

Comment:
Wow, not commenting on your code but this is a giant, giant oversight on my part.  A long stream of binary `1` on a UART indicates idle state between transmissions, not `0`.

This code, as it stands, will iterate over the entire idle period until it finds either the start of the buffer or the last `0` from the previous transmission.
An older version of the code used to grind to a halt when you disconnected the UART wire.  The "fix" was to detect this state and disable the decoder through the `wireDisconnected` signal.

A `while` loop is also not the best way to handle this, as it won't give up in the case of an invalid signal.  I think I'll need to have another look at this method when I find the time.